test/unit/v1/test_bucket.py

```diff
@@ -825,6 +894,69 @@ def _make_data(self, approximate_length):
         return b''.join(fragments)
 
 
+class TestConcatenate(TestCaseWithBucket):
+    def _create_remote(self, sources, file_name, encryption=None):
+        self.bucket.concatenate(sources, file_name=file_name, encryption=encryption)
+
+    def test_create_remote(self):
+        data = b'hello world'
+
+        f1_id = self.bucket.upload_bytes(data, 'f1').id_
+        f2_id = self.bucket.upload_bytes(data, 'f1').id_
+        with TempDir() as d:
+            path = os.path.join(d, 'file')
+            write_file(path, data)
+            self._create_remote(
+                [
+                    CopySource(f1_id, length=len(data), offset=0),
+                    UploadSourceLocalFile(path),
+                    CopySource(f2_id, length=len(data), offset=0),
+                ],
+                file_name='created_file'
+            )
```

Comment:
you should check if the resulting file is what you wanted it to be, otherwise it's a bad test